mne_bids/write.py

```diff
@@ -1352,3 +1353,163 @@ def write_anat(bids_root, subject, t1w, session=None, acquisition=None,
+
+    Parameters
+    ----------
+    ch_names : str | list of str
+        The names of the channel(s) to mark as bad. Pass an empty list in
+        combination with ``overwrite=True`` to mark all channels as good.
+    descriptions : None | str | list of str
+        Descriptions of the reasons that lead to the exclusion of the
+        channel(s). If a list, it must match the length of ``ch_names``.
+        If ``None``, no descriptions are added.
+    bids_basename : BIDSPath | str
+        The base filename of the BIDS compatible files. Typically, this can be
+        generated using :func:`mne_bids.make_bids_basename`.
+    bids_root : str | pathlib.Path
+        Path to root of the BIDS folder.
+    kind : str | None
+        The kind of recording to update. If ``None`` and only one kind (e.g.,
+        only EEG or only MEG data) is present in the dataset, it will be
+        selected automatically.
+    overwrite : bool
+        If ``False``, only update the information of the channels passed via
+        ``ch_names``, and leave the rest untouched. If ``True``, update the
+        information of **all** channels: mark the channels passed via
+        ``ch_names`` as bad, and all remaining channels as good, also
+        discarding their descriptions.
+    verbose : bool
+        The verbosity level.
+
+    Raises
+    ------
+    ValueError
+        If the numbers of channels and descriptions do not match.
+
+    ValueError
+        If the specified channels cannot be found in the dataset.
+
+    ValueError
+        If the specified ``kind`` cannot be found in the dataset.
+
+    ValueError
+        If ``overwite=False``, but no channel names were passed.
+
+    RuntimeError
+        If multiple recording kinds are present in the dataset, but
+        ``kind=None``.
+
+    RuntimeError
+        If more than one data files exist for the specified recording.
+
+    Examples
+    --------
+    Mark a single channel as bad.
+
+    >>> mark_bad_channels('MEG 0112', bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark multiple channels as bad.
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and add a description as to why.
+
+    >>> ch_names = ['MEG 0112', 'MEG 0131']
+    >>> descriptions = ['Only produced noise', 'Continuously flat']
+    >>> mark_bad_channels(bads, descriptions, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark channels as bad, and reset all others existing in the dataset to
+    "good".
+
+    >>> bads = ['MEG 0112', 'MEG 0131']
+    >>> mark_bad_channels(bads, overwrite=True, bids_basename=bids_basename,
+                          bids_root=bids_root)
+
+    Mark all channels as good.
+
+    >>> mark_bad_channels([], bids_basename=bids_basename, bids_root=bids_root)
```

Comment:
wouldn't you need overwrite=True here?

Reply:
Absolutely, I've addressed this in ad9f1ea